### PR TITLE
pkg/lightning: checkVersion checks major version for requiredMaxVersion (#806)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #806 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In br repo:
git pr https://github.com/pingcap/br/pull/854
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/br.git pr/854:release-4.0-3833d43edb1e
```

---

### What problem does this PR solve? 

Compare the major version number to make sure beta version does not pass
the check. This is because beta version may contains incompatible
changes.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### Release Note

 - No release note (it's not released yet.)

### CODE REVIEW

File: pkg/lightning/backend/checkreq_test.go [hunk 1]
```diff
@@ -94,7 +100,18 @@ func (s *checkReqSuite) TestCheckPDVersion(c *C) {
...
+=======
+	version = `{
+    "version": "v6.0.0-beta"
+}`
+	c.Assert(checkPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), ErrorMatches, "PD version too new.*")
+
+	version = `{
+    "version": "v1.0.0"
+}`
+>>>>>>> 3833d43... pkg/lightning: checkVersion checks major version for requiredMaxVersion (#806)
```
Comment (kennytm) [inline pkg/lightning/backend/checkreq_test.go:114]: merge conflict

**Review by overvenus — APPROVED** [review]

**Pull request merged**